x-pack/plugins/infra/public/common/visualizations/lens/hosts/cpu.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type Filter, FilterStateStore } from '@kbn/es-query';
+import type {
+  FormBasedLayer,
+  FormulaPublicApi,
+  PersistedIndexPatternLayer,
+  XYState,
+} from '@kbn/lens-plugin/public';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { SavedObjectReference } from '@kbn/core-saved-objects-common';
+import {
+  DEFAULT_LAYER_ID,
+  getBreakdownColumn,
+  getHistogramColumn,
+  getXYVisualizationState,
+} from '../utils';
+import type { LensOptions } from '../../../../types';
+import type { ILensVisualization } from '../types';
+
+const BREAKDOWN_COLUMN_NAME = 'hosts_aggs_breakdown';
+const HISTOGRAM_COLUMN_NAME = 'x_date_histogram';
```

Comment:
**nit:** These two constants are defined for each visualization config, they could be abstracted into an Enum/Object and be reused across all the classes and also to make stricter the type signature for `getBreakdownColumn` and `getHistogramColumn` for the expected string argument.

Reply:
Each visualization can define their own column names. This isn't necessarily something that needs to be strictly typed.